### PR TITLE
Added support for receiving HTML input and bug fix

### PR DESCRIPTION
**Pull request by advance512**
Added support for receiving HTML input
Fixed issue with dimensions(tag) not working for some websites

### CODE REVIEW

File: tests/test_favicon.py [hunk 1]
```diff
@@ -201,3 +201,16 @@ def test_request_kwargs(m):
...
...ific-Rim-300x76.png 300w, http://mock.com/wp-content/uploads/2013/07/Wide-Pacific-Rim-1024x258.png 1024w" sizes="(max-width: 1150px) 100vw, 1150px" /></a></p> </div> </div> </div><!-- /.main --> </div><!-- /.row--> </div><!-- /.content --> </div><!-- /.wrap --> <footer id="containerfooter" class="footerclass" role="contentinfo"> <div class="container"> <div class="row"> </div> <div class="footercredits clearfix"> <p>&copy; 2020 MOCK.com</p> </div> </div></footer><script type='text/javascript' src='http://mock.com/wp-content/plugins/responsive-image-maps/jquery.rwdImageMaps.min.js?ver=1.5'></script><script type='text/javascript' src='http://mock.com/wp-includes/js/imagesloaded.min.js?ver=3.2.0'></script><script type='text/javascript' src='http://mock.com/wp-includes/js/masonry.min.js?ver=3.3.2'></script><script type='text/javascript' src='http://mock.com/wp-content/themes/virtue/assets/js/min/plugins-min.js?ver=255'></script><script type='text/javascript' src='http://mock.com/wp-content/themes/virtue/assets/js/main.js?ver=255'></script><script type='text/javascript' src='http://mock.com/wp-includes/js/wp-embed.min.js?ver=5.1.1'></script> </div><!--Wrapper--> </body></html>''' # noqa
```
Comment (scottwernervt) [inline tests/test_favicon.py]: Remove unnecessary html that is not related to the test. For example, the icons should only come from `<head>` and not the contents of `<body>`. Keep it simple :)

```
<html>
   <head>
      icons to look for here
   </head>
   <body>
       <p>Test</p>
   </body>
</html>

```

File: tests/test_favicon.py [hunk 2]
```diff
@@ -201,3 +201,16 @@ def test_request_kwargs(m):
...
+
+    icons = favicon.get(
+        'http://mock.com/',
+        html=mock_com_html,
+    )
+    assert icons
+    assert icons[0].url == 'http://mock.com/favicon.ico'
+    assert icons[1].url == 'http://mock.com/wp-content/uploads/2014/03/favicon.ico'
+    print(icons)
```
Comment (scottwernervt) [inline tests/test_favicon.py]: Remove `print statement` and at line break to end of file.

File: src/favicon/favicon.py [hunk 3]
```diff
@@ -40,12 +40,14 @@
...
 
-def get(url, *args, **request_kwargs):
+def get(url, *args, html=None, **request_kwargs):
     """Get all fav icons for a url.
 
     :param url: Homepage.
     :type url: str
 
+    :param html: HTML input, as string. Will be used instead of an HTTP response from the `url`.
+
```
Comment (scottwernervt) [inline src/favicon/favicon.py:51]: 1. Can you rename `html` to `html_override` so it is clear why a user would pass a value to it?
2. Add `:type html: str or None`

**Review by scottwernervt — CHANGES_REQUESTED** [review]
I am not sure why the Travis CI is not showing up in gitlab checks but there are some errorshttps://travis-ci.org/github/scottwernervt/favicon/builds/699454421.

The py27 test is failing so I think we might have to move `html_override` arg in front of `*args`.

**Comment by advance512** [comment]
> I am not sure why the Travis CI is not showing up in gitlab checks but there are some errorshttps://travis-ci.org/github/scottwernervt/favicon/builds/699454421.
> 
> The py27 test is failing so I think we might have to move `html_override` arg in front of `*args`.

Unfortunately we can't do that, as that will mean that code like this:
```
favicon.get('http://mock.com/', headers)
```
will see the headers as the `html` parameter and break. Putting the `html` parameter after the `args*` means it becomes a keyword only argument, which solves it.
Alternatively, we can modify the version to show non-comaptibility, or drop Python 2.x support (it is EOL'd after all).

File: tests/test_favicon.py (hunk 1 above)
Comment (advance512) [reply tests/test_favicon.py]: While some unit tests test edge cases or specific situation, I wanted this test to simulate real-life as best we can. Anyways, I removed the extra HTML.

File: tests/test_favicon.py (hunk 2 above)
Comment (advance512) [reply tests/test_favicon.py]: Done

File: src/favicon/favicon.py (hunk 3 above)
Comment (advance512) [reply src/favicon/favicon.py:51]: Done

File: tests/test_favicon.py (hunk 1 above)
Comment (advance512) [reply tests/test_favicon.py]: Done
